### ISSUE TITLE
Can't open a file downloaded by DropBox

### ISSUE THREAD

**Issue by limelect** [IsraelHikingMap/Site#1246]
## Bug
ניסיתי להעלות מ DROPBOX קובץ לא עולה.
אותו הקובץ עולה מ GOOGLE DRIVE
Android s6

**Comment by zstadler** [comment]
To reproduce:
1. Upload a `gpx` file to Dropbox. It could be a `gpx` file created by saving a route planned by the app.
2. Open the Dropbox app
3. Tap on the `gpx` file

I would expect to see the "Israel Hiking Map" app in the list, but it is not there.

![image](https://user-images.githubusercontent.com/1304610/84793108-66880b80-affd-11ea-87c4-6fa1fe1fd6dd.png)

**Comment by limelect** [comment]
No.
Open >Track>Open from>Choos DropBox > Go To directory (Choose file in)> Press file
>Download

Go back to the application. Nothing. Hop it clarify.

**Comment by zstadler** [comment]
@limelect,
Please attach screenshots for the first 3 steps: 
- Open
- Track
- Open from

**Comment by limelect** [comment]
I made a movie where to send it is big

**Comment by limelect** [comment]
I have 2 movies for you. Plz communicate with my mail

**Comment by HarelM** [comment]
Video summary:
1. Click open file
2. Select dropbox from sidebar menu
3. Select a file in a dropbox folder (inside the dropbox app)
4. The file finishes to download
5. Dropbox exits
--> Nothing happens - the app doesn't receive the file selection.

I have no clue how to fix this.
I have commented in the below issue, I hope someone will know better...
https://github.com/apache/cordova-android/issues/816

**Comment by limelect** [comment]
Harel can you check plz if a "NORMAL" file in any directory is loaded?  Not YOUR directory.
Then in the source debug you do not get the file?
I would have helped if I hade the souces.

**Comment by HarelM** [comment]
Opening regular files in any folder is working.
The sources are here in this repository, I'd be happy to walk you through on how to create an environment to debug this issue.
The main problem I believe is that this is using an abstraction layer called Cordova which I believe is what causing this issue (see linked issue above) but I'm not sure...

When doing the same from the browser it works so it might be a difference between how the browser works and how Cordova implemented the file picker, again, I'm not sure.

Any help would be appreciated :-)

**Comment by limelect** [comment]
See all my personal mails that might help.
To overcome the problem

**Comment by HarelM** [comment]
Not sure I understood the solution you wrote...
Fixing the file association using another app solved this issue?
i.e. When setting the app to open gpx files using Astro fixes this issue of selecting a file from  dropbox when clicking the open file from the IHM app?

**Comment by limelect** [comment]
No, if it is from within your application.
However, with a file manager, it is opened with your application.
What I am trying to find is why YOUR application is not on the list
as other maps drawing apps are.

**Comment by HarelM** [comment]
This is due to Android's nightmare of file association mechanism.
All the attempts to handle this can be found here:
https://github.com/IsraelHikingMap/Site/blob/master/IsraelHiking.Web/config.xml#L30L130
>What I am trying to find is why YOUR application is not on the list

What list? where?

>as other maps drawing apps are when opening the file from within your application.

With in IHM application when opening a file? this line doesn't make any sense...

**Comment by limelect** [comment]
Can this be a possible fix?
https://stackoverflow.com/questions/46317586/my-app-is-not-showing-in-the-list-to-open-txt-file-in-android-7-1-1
<data android:scheme="content"
          android:mimeType="text/*" />
<data android:mimeType="*/*" />

**Comment by HarelM** [comment]
It might, I don't know... I need to use logcat to see which intent is sent and hopefully with what properties and add the right line.
Mainly trial and error, much like previous attempts to solve issue like that...

**Comment by limelect** [comment]
Well, good luck. I hope i did my part

**Comment by HarelM** [comment]
Thanks for the info! 
Feel free to submit a PR if you want this sooner than later :-)

**Comment by HarelM** [comment]
I've added a fix to the dropbox file association issue, i.e. when opening the dropbox app and selecting a file the IHM should now be part of the list of options for opening this file.
This doesn't help with the scenario of opening a file from inside the IHM app, selecting dropbox from the sidebar and then selecting a file. Apparently this issue also exists with the regular file manager app which is suppose to be more simple, and I think this is an android-cordova bug. I'll see if upgrading to latest version solves this somehow. If not I don't know how to proceed here and I'll probably close this bug as "wontfix"...

**Comment by zstadler** [comment]
I've tested the build and it works as described above.
I also found a workaround for opening a Dropbox file from the application: 

After select the required file from the "Dropbox" category, wait for "Downloading" to complete and **select the required file again**. 

Notes:
- I believe different manufacturers are using different file selectors. Hopefully the workaround is not specific to this file selector.
- It is not clear to me if this is a bug of the file selector or a bug in the Dropbox app
